geomstats/geometry/matrices.py

```diff
@@ -329,6 +329,52 @@ def trace_product(mat_1, mat_2):
         return gs.einsum(
             '...ij,...ji->...', mat_1, mat_2)
 
+    def math_vec(self, mat):
+        """Return a vectorized form of the matrix.
```

Comment:
Maybe use `flat` or `1d` instead of vectorized as this is not the way this word is used in the codebase. @ninamiolane what do you think?

Reply:
I agree, maybe "Flatten a matrix into a vector."?

Reply:
I am against 1d since a '1d' vector can have an arbitrary dimension larger than 1... In tensor language, on would talk about 1-way, 2-way or n-way tensors rather than 1d or 2d. Moreover, the expected result on an array of matrices would actually be a 2d numpy array...

"Flatten" could be ok, but this is also the naming used in numpy with `ndarray.flatten()` for which the expected result on an array of matrices would be a single array with all values, while I want here to obtain the array of flatten matrices. Actually maybe the term `vectorized_flatten` could work. There needs to be a pending name for the reverse operation (`vectorized_reshape` or `vectorized_unfolding`?).

However, I am uncomfortable with the computer science restriction of the meaning of the mathematical word vectorization to only a subset of its original mathematical meaning. I don't see why vectorization should only concern the data axis0 and not the other axes. I think that we should be careful with this.  

In order to ground the discussion, a useful reference to tensor reshaping operations and their names is [the wikipedia page Tensor_reshaping](https://en.wikipedia.org/wiki/Tensor_reshaping).

Reply:
I have renamed the methods flatten / reshape and edited the docstrings (+ added a test on dimension for reshaping).